vcd/resource_vcd_catalog.go

```diff
@@ -401,41 +461,62 @@ func resourceVcdCatalogImport(_ context.Context, d *schema.ResourceData, meta in
 	return []*schema.ResourceData{d}, nil
 }
 
-func createOrUpdateAdminCatalogMetadata(d *schema.ResourceData, meta interface{}) error {
-
-	log.Printf("[TRACE] adding/updating metadata for catalog")
-
-	vcdClient := meta.(*VCDClient)
-
-	org, err := vcdClient.GetAdminOrgFromResource(d)
-	if err != nil {
-		return fmt.Errorf(errorRetrievingOrg, err)
-	}
-
-	catalog, err := org.GetAdminCatalogByName(d.Get("name").(string), false)
-	if err != nil {
-		log.Printf("[DEBUG] Unable to find catalog.")
-		return fmt.Errorf("unable to find catalog: %s", err)
-	}
-
-	return createOrUpdateMetadata(d, catalog, "metadata")
-}
-
-func setCatalogData(d *schema.ResourceData, adminOrg *govcd.AdminOrg, catalogName string) error {
+func setCatalogData(d *schema.ResourceData, adminOrg *govcd.AdminOrg, adminCatalog *govcd.AdminCatalog, resourceType string) error {
 	// Catalog record is retrieved to get the owner name, number of vApp templates and medias, and if the catalog is shared and published
-	catalogRecords, err := adminOrg.FindCatalogRecords(catalogName)
+	catalogRecords, err := adminOrg.FindCatalogRecords(adminCatalog.AdminCatalog.Name)
 	if err != nil {
 		log.Printf("[DEBUG] Unable to retrieve catalog record: %s", err)
 		return fmt.Errorf("unable to retrieve catalog record - %s", err)
 	}
 
 	dSet(d, "catalog_version", catalogRecords[0].Version)
 	dSet(d, "owner_name", catalogRecords[0].OwnerName)
-	dSet(d, "number_of_vapp_templates", catalogRecords[0].NumberOfVAppTemplates)
-	dSet(d, "number_of_media", catalogRecords[0].NumberOfMedia)
 	dSet(d, "is_published", catalogRecords[0].IsPublished)
 	dSet(d, "is_shared", catalogRecords[0].IsShared)
 	dSet(d, "publish_subscription_type", catalogRecords[0].PublishSubscriptionType)
 
+	var rawMediaItemsList []interface{}
+	var rawVappTemplatesList []interface{}
+
+	var mediaItemList []string
+	var vappTemplateList []string
+
+	mediaItems, err := adminCatalog.QueryMediaList()
+	if err != nil {
+		return fmt.Errorf("%v", err)
+	}
+
+	for _, media := range mediaItems {
+		mediaItemList = append(mediaItemList, media.Name)
+	}
+
+	vappTemplates, err := adminCatalog.QueryVappTemplateList()
+	if err != nil {
+		return fmt.Errorf("%v", err)
+	}
+	for _, vappTemplate := range vappTemplates {
+		vappTemplateList = append(vappTemplateList, vappTemplate.Name)
+	}
+
+	// Sort the lists, so that they will always match in state
+	sort.Strings(mediaItemList)
```

Comment:
Just curiosity, could this be an unordered TypeSet instead?

Reply:
It could, but I chose the list for two reasons:
1. easier to handle in code
2. easier to look through when many items are in the list. A set would make it practically unreadable for humans.